Tools/node/scripts/common.js

```diff
@@ -0,0 +1,67 @@
+const path = require('path')
+const fs = require('fs');
+const child_process = require('child_process')
+const supportedCpuArchitectures = ['x64', 'arm64'];
+const supportedPlatforms = ['win32', 'darwin', 'linux']
+
+
+/**
+ * Ensures the current host OS is supported by the Bebop compiler
+ * @param {NodeJS.Architecture} arch the host arch
+ * @param {NodeJS.Platform} platform  the host os
+ */
+function ensureHostSupport(arch, platform) {
+    if (!supportedCpuArchitectures.includes(arch)) throw new Error(`Unsupported CPU arch: ${arch}`);
+    if (!supportedPlatforms.includes(platform)) throw new Error(`Unsupported platform: ${platform}`);
+}
+/**
+ * Gets information about the current compiler host
+ * @returns 
+ */
+function getHostInfo() {
+    const arch = process.arch;
+    const platform = process.platform;
+    ensureHostSupport(arch, platform);
+    const osName = () => {
+        switch (platform) {
+            case "win32":
+                return "windows";
+            case "darwin":
+                return "macos";
+            case "linux":
+                return "linux";
+            default:
+                throw new Error(`Unknown platform name: ${platform}`);
+        }
+    };
+    return { arch: arch, os: osName, exeSuffix: osName === "windows" ? '.exe' : '' };
```

Comment:
Hang on, I think you made a mistake. `osName` will be a function here, not a string. You would need to turn that into an immediately evaluated function expression by wrapping it like `const osName = (() => { ... })()`